src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void ensureCapacity() {
+        if (size == elements.length) {
+            int newCapacity = (int) (elements.length * 1.5);
+            elements = Arrays.copyOf(elements, newCapacity);
```

Comment:
![image](https://github.com/mate-academy/jv-array-list/assets/20326858/eba4e6a6-13f0-4ca2-a0f9-6e576a7e1b17)

Reply:
sorry for my inattention